src/backend/Operators.js

```diff
@@ -0,0 +1,47 @@
+const BinaryOperator = require('./BinaryOperator');
+
+class Addition extends BinaryOperator {
```

Comment:
This is fine as is. But we should keep in mind where the actual math will be done. In these classes or elsewhere? And would it be possible to move this constructor to the BinaryOperator class as they are all the same?

Reply:
This class is actually NOT right IMO. It doesn't make sense to pass left and right to the parent class (where they are stored as "this.leftOperand" and "this.rightOperand" AND store them in this class when Addition extends (and therefor inherits the members from) BinaryOperator

Reply:
As for where the math should be, why not add a "reduce" method to the Node object which will be implemented at this level (i.e. not in BinaryOperator which is abstract). Basically OOP parse tree reduction just like in SER502. I don't see a reason to do it outside the Node class since it would make it pretty easy. (Just like getText but instead of returning "5 + 3" it returns 8